packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1523,6 +1559,16 @@ export class TransactionController extends BaseController<
       pendingTxs,
     );
 
+    // Make sure provided external transaction has non empty history array
+    if (
+      !Array.isArray(transactionMeta.history) ||
```

Comment:
Could this be simplified to `!(transactionMeta.history ?? []).length`?

Reply:
That's true, it could be simplified to your suggestion.
But to be honest, I couldn't say that history is expected as an array at first look. Isn't it more readable with the current code?

Reply:
Actually there is also a comment ....

Reply:
Leaving the comment, updated the condition here: https://github.com/MetaMask/core/pull/1657/commits/ca08c4dbb561775bd67f25a4405c29e4c61d4fd3